### PR TITLE
And/But support for Rspec-Given

### PR DESCRIPTION
**Pull request by christopherdolan**
Hi Jim,

I was a student in your TDD class this Friday.  For my last lab, I decided to work on And & But support for Rspec-Given.

To that end, I've made modifications to the syntax of your stack tests to utilize these conditional aliases.  It works in those cases.  Please let me know what you think!

Thanks,
~Chris Dolan

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -204,6 +205,14 @@ assertion. Code in _Then_ blocks should not have any side effects.
 After the related _When_ block is run, the stack should be empty.  If
 it is not empty, the test will fail.
 
+#### And/But:
+
+_And_ and _But_ are syntactic sugar that essentially serve to continue
+the previous expression. As there can only be one _When_ for a given test,
+_And_ and _But_ only worko following _Given_ and _Then_.  There are also
```
Comment (christopherdolan) [inline README.md:212]: typo!  'worko'... really?

**Comment by dnagir** [comment]
This pull request looks nice. I wonder if somebody wants to actually merge it in?

**Comment by searls** [comment]
I've actually had a brief chat with Jim on this topic recently. See, I implemented `And` in [jasmine-given](https://github.com/searls/jasmine-given) and upon boasting about it, Jim quickly cooled my enthusiasm with several entirely valid criticisms of including And (and presumably, But) in a GWT testing DSL.

I'd rather @jimweirich restate the case here if he doesn't mind, as I feel like I've already put too many words in his mouth on the matter.

**Comment by christopherdolan** [comment]
I'd love to hear it.  Honestly, I have no emotional connection to this patch.  I was in a class that Jim was teaching, got curious about a project he described, and decided to hack on it for the class' final lab.  I was much more concerned with the technical implications of including this support than whether it was appropriate to do so.

**Comment by jimweirich** [comment]
I gave Justin 4 reasons why I don't care for the And/But syntax.  Let's see if I can remember them.

(1) And implies sequential processing or some kind of ordering.  The lazy givens can actually be executed in any order.

(2) I tend to move Givens up and down the nested contexts as I refactor.  If I move the first Given, then I have to also change the following And to a Given.

(3) The impulse to use And is a product of the trend of using faux-english for DSLs. Although I like readable code, I don't insist the code must be grammatically correct English.  Some of rspec's faux-english constructs are actually ambiguous, so are FAIL w.r.t. specifications.

(4) The Ands are ambiguous.  I can no longer look at a line of code to understand what it does, but have to visually scan upwards to see if it follows a Given or a Then.

I suspect that if I had implemented And/But early in the rspec-given lifecycle, I would not have discovered some of the nice patterns of interaction that happen when you place your Givens/Whens/Thens at different context levels.

All in all, I don't think the drawbacks to And are overwhelming, but given that the benefits are minuscule I am not planning on adding And/But to respec-given.

But I totally get writing the patch for the technical chops.  More power to you!  And thanks for the contribution.

**Comment by searls** [comment]
Thanks @jimweirich! For your trouble, I just removed all references to `And` from jasmine-given's code examples. https://github.com/searls/jasmine-given/commit/8ea1cfc670f94e1f91028d95782f83997e793cbb

**Pull request closed without merging**